### PR TITLE
token extractor utils and url validation

### PR DESCRIPTION
**Pull request by sym3tri**
This obviates some of the token extraction boilerplate when using this library.
Also adds stricter URL validation which was causing uncaught errors to bubble up during actual usage

### CODE REVIEW

**Comment by bobbyrullo** [comment]
minor things, otherwise LGTM

**Comment by sym3tri** [comment]
addressed comments and rebased

File: oidc/client.go [hunk 1]
```diff
@@ -66,11 +66,16 @@ type ClientConfig struct {
 }
 
 func NewClient(cfg ClientConfig) (*Client, error) {
+	ru, err := phttp.ParseNonEmptyURL(cfg.RedirectURL)
```
Comment (yifan-gu) [inline oidc/client.go]: @sym3tri Later for the party, but why redirectURL is required here? E.g. I think I can use the client for just token verification or token refreshing, which doesn't need the redirectURL info.

Comment (sym3tri) [inline oidc/client.go]: Clients need a callback URL in order to get tokens from the OIDC server. That's the only way a token can be obtained. Why would a client ever need to validate a token without allowing it to be generated?

Are there any actual use-cases requiring verification without generation? Even if a user is pasting in a token + refresh token, I think the client still needs to support logging in via the browser and the 3-legged OAuth2 flow if available.

Comment (yifan-gu) [inline oidc/client.go]: @sym3tri My use case here is the kube-apiserver acts as a client, but it doesn't handle login or cause to generate any tokens, it just verifies the token.

Comment (sym3tri) [inline oidc/client.go]: I see... that makes sense. Feel free to remove this validation if it helps your case.

Comment (yifan-gu) [inline oidc/client.go]: Good to know, thanks :)

**Pull request merged**